charts/rhio/templates/deployment.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "app.fullname" . }}
+  labels:
+    {{- include "app.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "app.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "app.labels" . | nindent 8 }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
```

Comment:
I don't see how configuration file is passed...
 
I think you need to define config file section in values.yaml. Then save this section in the configmap by helm template and then mount the file from the configmap to container file system. This is for the configuration of subscriptions, publishers and other configuration parameters.

For the private key i think you need a separate secret. Since it is defined by the user you probably want to make an entry in values.yaml where user can set that key. The key goes to a secret and from there mounted as a file to container fs.